### PR TITLE
Implement comparison methods in Sku class and add its tests

### PR DESCRIPTION
**Pull request by YalinLi0312**
The Sku class properties comparison priority I followed: Name > Family > Size > Tier > Capacity

Code coverage result in VisualStudio:
Screenshot 2020-11-18 154916

Code coverage result in Cobertura:
Screenshot 2020-11-18 155304

### CODE REVIEW

File: azure-proto-core-test/SkuTests.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
+﻿using azure_proto_core;
+using NUnit.Framework;
+
+namespace azure_proto_core_test
+{
+    class SkuTests
+    {
+        [Test]
+        public void CompareToTests()
```
Comment (m-nash) [inline azure-proto-core-test/SkuTests.cs:9]: Break these out into tests for each parameter and use parameterized testing like we discussed in our sync today

File: azure-proto-core-test/SkuTests.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        {
+            Sku sku1 = new Sku();
+            Sku sku2 = null;
+
+            sku1.Name = "name";
```
Comment (m-nash) [inline azure-proto-core-test/SkuTests.cs:14]: Verify with @markcowl where the constraints are documented for this object?  It is possible this is deserialize only in which case that is "irrelevant"?

**Review by m-nash — CHANGES_REQUESTED** [review]

File: azure-proto-core-test/SkuTests.cs (hunk 1 above)
Comment (YalinLi0312) [reply azure-proto-core-test/SkuTests.cs]: Updated

File: azure-proto-core-test/SkuTests.cs (hunk 2 above)
Comment (YalinLi0312) [reply azure-proto-core-test/SkuTests.cs]: Pending until getting the exact constraints from the contact. Will update in a separate PR if need to.

**Comment by YalinLi0312** [comment]
New code coverage results:
Screenshot 2020-11-20 121344
Screenshot 2020-11-20 121426

**Comment by YalinLi0312** [comment]
Update code coverage results:
Screenshot 2020-11-20 163215
Screenshot 2020-11-20 163259

File: azure-proto-core/Resources/Sku.cs [hunk 3]
```diff
@@ -17,12 +16,30 @@ public class Sku : IEquatable<Sku>, IComparable<Sku>
 
         public int CompareTo(Sku other)
         {
-            throw new NotImplementedException();
+            if (other == null) return 1;
+            if (this.Name == null && other.Name != null) return -1;
+            if (this.Name != null && this.Name.CompareTo(other.Name) != 0) return this.Name.CompareTo(other.Name);
```
Comment (m-nash) [inline azure-proto-core/Resources/Sku.cs:21]: store this.Name.CompareTo(other.Name) so you don't have to call it twice

this applies to all the other items in this method and Equals

**Review by m-nash — CHANGES_REQUESTED** [review]

**Review by bquantump — APPROVED** [review]

**Pull request merged**